src/lib/snark_params/scan_state_constants.ml

```diff
@@ -0,0 +1,46 @@
+[%%import
+"../../config.mlh"]
+
+(* in the config files, we either have:
+
+    - transaction_capacity_log_2 and work_delay exp explicitly, or
+    - those are calculated from the tps_goal and num_snark_workers
+
+   use the config boolean scan_state_calc_params to distinguish those cases
+
+ *)
+
+module type S = Scan_state_constants_intf.S
+
+[%%if
+scan_state_calc_params]
+
+open Core_kernel
+
+[%%inject
+"tps_goal", scan_state_tps_goal]
+
+[%%inject
+"num_snark_workers", scan_state_num_snark_workers]
+
+[%%inject
+"block_window_duration", block_window_duration]
+
+let max_coinbases = 2
+
+let max_user_commands_per_block = tps_goal * block_window_duration
+
+let transaction_capacity_log_2 =
+  1 + Int.ceil_log2 (max_user_commands_per_block + max_coinbases)
```

Comment:
I think there will be at least one fee transfer. Maybe there can be zero if the prover has the same public key as the proposer. If we want transactions per second to mean user commands per second then we should come up with an expected SNARK work public key "density" and estimate how many fee transfers there will be per block.

Reply:
This one's assuming unique prover for each proof in which case max # of fee transfers you could have is `(max_user_commands_per_block + max_coinbases) - 1`. Therefore, the transaction capacity should be big enough to include twice as many transactions as the number of user commands and coinbase transactions are included (which is part of the answer to the next question).